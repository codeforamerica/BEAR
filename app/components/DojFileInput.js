// @flow
import React, { Component } from 'react';
import styles from './DOJFileInput.css';

type Props = {
  onFileSelect: string => void
};

export default class DojFileInput extends Component<Props> {
  handleFileSelection = (event: SyntheticEvent<HTMLInputElement>) => {
    const { onFileSelect } = this.props;
    const filePath = event.currentTarget.files[0].path;
    onFileSelect(filePath);
  };

  render() {
    return (
      <div className="file-upload">
        <label className="button file-upload__label" htmlFor="doj-file-input">
          <input
            onChange={this.handleFileSelection}
            type="file"
            accept=".dat, .csv"
            id="doj-file-input"
            name="doj-file-input"
            className="file-upload__input"
          />
          Select file
        </label>
        <p className={styles.emptyMessage}>No file selected</p>
      </div>
    );
  }
}
