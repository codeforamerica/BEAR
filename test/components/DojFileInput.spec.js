import sinon from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import DojFileInput from '../../app/components/DojFileInput';

Enzyme.configure({ adapter: new Adapter() });
const sandbox = sinon.createSandbox();

function setup() {
  const onFileSelectSpy = sandbox.spy();
  const component = shallow(<DojFileInput onFileSelect={onFileSelectSpy} />);
  return {
    component,
    onFileSelectSpy
  };
}

afterEach(() => {
  sandbox.restore();
});

describe('DojFileInput component', () => {
  it('selecting file should call onFileSelect with file path', () => {
    const { component, onFileSelectSpy } = setup();
    const fileInput = component.find('#doj-file-input');
    const fakeEvent = {
      currentTarget: {
        files: [{ path: 'path/to/file' }]
      }
    };
    fileInput.simulate('change', fakeEvent);
    expect(onFileSelectSpy.called).toBe(true);
    const { args } = onFileSelectSpy.getCall(0);
    expect(args[0]).toEqual('path/to/file');
  });

  it('should match exact snapshot', () => {
    const { component } = setup();
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
