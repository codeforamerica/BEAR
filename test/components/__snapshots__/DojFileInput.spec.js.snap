// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DojFileInput component should match exact snapshot 1`] = `
<div>
  <label
    className="noBottomMargin button file-upload__label"
    htmlFor="doj-file-input"
  >
    <input
      accept=".dat, .csv"
      className="file-upload__input"
      id="doj-file-input"
      multiple={true}
      name="doj-file-input"
      onChange={[Function]}
      type="file"
    />
    Select file
  </label>
</div>
`;
